app/(dashboard)/item/[id]/page.tsx

```diff
@@ -71,14 +72,17 @@ const DisplayItemDetails = async ({ params }: { params: { id: string } }) => {
               fontSize='text-lg'
             />
 
-            {canMessage && (
-              <NewConversationButton
-                userId={user?.id}
-                donorId={donerId}
-                donorEmail={donorEmail}
-                title={title}
-                item_id={item.id}
-              />
+            {canMessage && user?.id && item.id && (
+              <div className='flex flex-row gap-4 pb-10'>
+                <NewConversationButton
+                  userId={user?.id}
+                  donorId={donerId}
+                  donorEmail={donorEmail}
+                  title={title}
+                  item_id={item.id}
+                />
+                <ReserveButton itemId={item.id} userId={user.id} />
+              </div>
             )}
```

Comment:
`canMessage` already checks for us that there is a user and item so adding there here adds back the complexity that we originally abstracted. Let's keep the evaluation just as it was and just tell typescript that we know user is not `undefined`

```typescript
 {canMessage && (
              <div className='flex flex-row gap-4 pb-10'>
                <NewConversationButton
                  userId={user?.id}
                  donorId={donerId}
                  donorEmail={donorEmail}
                  title={title}
                  item_id={item.id}
                />
                <ReserveButton itemId={item.id} userId={user?.id!} />
              </div>
```

Reply:
Done